tests/test_structures.py

```diff
@@ -74,3 +74,97 @@ def test_getitem(self, key, value):
+        assert self.cache['a'] is self.any_value
+
+    def test_repr(self):
+        repr = str(self.cache)
+        assert repr == '<timed-cache maxlen:32 len:0>'
+
+    def test_get_expired_item(self, mocker):
+        self.cache = TimedCache(maxlen=1, expiration_secs=self.expiration_secs)
+
+        mocker.patch('requests.structures.current_time', lambda: 0)
+        self.cache['a'] = self.any_value
+        mocker.patch('requests.structures.current_time', lambda: self.expiration_secs + 1)
+        assert self.cache.get('a') is None
+
+    def test_evict_first_entry_when_full(self, mocker):
+        self.cache = TimedCache(maxlen=2, expiration_secs=2)
+        mocker.patch('requests.structures.current_time', lambda: 0)
+        self.cache['a'] = self.any_value
+        mocker.patch('requests.structures.current_time', lambda: 1)
+        self.cache['b'] = self.any_value
+        mocker.patch('requests.structures.current_time', lambda: 3)
+        self.cache['c'] = self.any_value
+        assert len(self.cache) is 2
+        with pytest.raises(KeyError, message='Expected key not found'):
+            self.cache['a']
+        assert self.cache['b'] is self.any_value
+        assert self.cache['c'] is self.any_value
+
+    def test_delete_item_removes_item(self):
+        self.cache['a'] = self.any_value
+        del self.cache['a']
+        with pytest.raises(KeyError, message='Expected key not found'):
+            self.cache['a']
+
+    def test_iterating_hides_timestamps(self):
+        self.cache['a'] = 1
+        self.cache['b'] = 2
+        expected = [('a', 1), ('b', 2)]
+        actual = [(key, val) for key, val in self.cache]
+        assert expected == actual
+
+
+class TestTimedCacheManagedDecorator(object):
+
+    def test_caches_repeated_calls(self, mocker):
+
+        preserved = TimedCacheManaged.__init__
+        nonlocals = {}
+        def replace_instance(slf, _):
+            fnc_stub = mocker.stub(name='fnc stub')
+            preserved(slf, fnc_stub)
+            nonlocals['cache'] = slf.cache
+            nonlocals['fnc_stub'] = fnc_stub
+
+        mocker.patch('requests.structures.TimedCacheManaged.__init__', replace_instance)
+        invocation_spy = mocker.spy(TimedCacheManaged, '__call__')
+        getitem_spy = mocker.spy(TimedCache, '__getitem__')
+        setitem_spy = mocker.spy(TimedCache, '__setitem__')
+
+        @TimedCacheManaged
+        def some_method(x):
+            return x + 1
+
+        cache = nonlocals['cache']
+        fnc_stub = nonlocals['fnc_stub']
+
+        assert len(cache) is 0
+        some_method(1)
+        assert getitem_spy.call_count is 1
+        assert setitem_spy.call_count is 1
+        assert fnc_stub.call_count is 1
+
+        assert len(cache) is 1
+        some_method(1)
+        assert getitem_spy.call_count is 2
+        assert setitem_spy.call_count is 1
+        assert fnc_stub.call_count is 1
+        assert len(cache) is 1
+
+        assert invocation_spy.call_count is 2
```

Comment:
Heh, so this is a good test, but it mostly tests implementation details. For this, it's enough to just mock out the time and then write a small function that constantly returns an incrementing value, then wrap it in the decorator and use the timer to demonstrate the behaviour.

Generally speaking, avoiding mocks is a good thing wherever we can. =)